core/views/browse.py

```diff
@@ -172,21 +173,28 @@ def issue_pages_rdf(request, lccn, date, edition):
 
 @cache_page(settings.DEFAULT_TTL_SECONDS)
 @vary_on_headers('Referer')
-def page(request, lccn, date, edition, sequence, words=None):
-    fragments = []
-    if words:
-        fragments.append("words=" + words)
-    qs = request.META.get('QUERY_STRING')
-    if qs:
-        fragments.append(qs)
-    if fragments:
-        path_parts = dict(lccn=lccn, date=date, edition=edition,
-                          sequence=sequence)
-        url = urlresolvers.reverse('chronam_page',
-                                   kwargs=path_parts)
-
-        return HttpResponseRedirect(url + "#" + "&".join(fragments))
+def page_words(request, lccn, date, edition, sequence, words=None):
+    """
+    for the case where we have ;words= in the url convert it to a fragment but
+    keep everything else the same so we don't mess up campain codes
+    
+    example:
+    /lccn/sn83045396/1911-09-17/ed-1/seq-12/;words=foo?bar=ham
+    becomes:
+    /lccn/sn83045396/1911-09-17/ed-1/seq-12/?bar=ham#words=foo
+
+    see https://github.com/LibraryOfCongress/chronam/issues/126
+    """
+    warnings.warn('url with ";words=" was called which is deprecated!', category=DeprecationWarning)
 
+    path_parts = dict(lccn=lccn, date=date, edition=edition, sequence=sequence)
+    url = urlresolvers.reverse('chronam_page', kwargs=path_parts)
+    redirect = "%s?%s#words=%s" % (url, request.GET.urlencode(), words)
```

Comment:
This should be urlencoded — I don't have push access to this repo but you can cherry-pick this commit: https://github.com/acdha/chronam/commit/cba7ae4a0a710f57aac5b90f6357570050991784